extensions/westFax/actions/sendFax/sendFax.ts

```diff
@@ -0,0 +1,135 @@
+import { isNil } from 'lodash'
+import { type Action } from '@awell-health/extensions-core'
+import { Category } from '@awell-health/extensions-core'
+import { type settings } from '../../settings'
+import { fields } from './config'
+import { dataPoints } from './config/dataPoints'
+
+export const sendFax: Action<typeof fields, typeof settings> = {
+  key: 'sendFax',
+  category: Category.COMMUNICATION,
+  title: 'SendFax',
+  description: 'Send fax in WestFax.',
+  fields,
+  dataPoints,
+  previewable: false,
+  onActivityCreated: async (payload, onComplete, onError): Promise<void> => {
+    const { fields, settings } = payload
+    const { product_id, feedback_email, number, content } = fields
+    try {
+      if (isNil(product_id) || isNil(number) || isNil(content)) {
+        await onError({
+          events: [
+            {
+              date: new Date().toISOString(),
+              text: { en: 'Fields are missing' },
+              error: {
+                category: 'MISSING_FIELDS',
+                message:
+                  '`product_id`, `fax_ids` or `feedback_email` is missing',
+              },
+            },
+          ],
+        })
+        return
+      }
+
+      const { username, password } = settings
+      if (isNil(username) || isNil(password)) {
+        await onError({
+          events: [
+            {
+              date: new Date().toISOString(),
+              text: { en: 'API requires an Username and password' },
+              error: {
+                category: 'MISSING_SETTINGS',
+                message: '`username` or `password` is missing',
+              },
+            },
+          ],
+        })
+        return
+      }
+
+      const blob = new Blob([content], { type: 'text/html' })
+
+      const formdata = new FormData()
+      formdata.append('Username', username)
+      formdata.append('Password', password)
+      formdata.append('Cookies', 'false')
+      formdata.append('ProductId', product_id)
+      if (!isNil(feedback_email)) {
+        formdata.append('FeedbackEmail', feedback_email)
+      }
+      formdata.append('Numbers1', number)
+      formdata.append('Files0', blob, 'content.html')
+
+      const requestOptions = {
+        method: 'POST',
+        body: formdata,
+      }
+
+      const response = await fetch(
```

Comment:
maybe use fetchTyped, it allows you to specify zod schema for the response to you get validated and typed response. Would make handling of the response below easier

Reply:
This is a post with formData, I couldn't understand how I could use fetchTyped with it. So I decide to not do it now.